execution/function/functions.go

```diff
@@ -355,6 +355,48 @@ var Funcs = map[string]FunctionCall{
 			},
 		}
 	},
+	"xrate": func(f FunctionArgs) promql.Sample {
+		if len(f.Points) == 0 {
+			return InvalidSample
+		}
+		v, h := extendedRate(f.Points, true, true, f.StepTime, f.SelectRange, f.Offset)
+		return promql.Sample{
+			Metric: f.Labels,
+			Point: promql.Point{
+				T: f.StepTime,
+				V: v,
+				H: h,
+			},
+		}
+	},
+	"xdelta": func(f FunctionArgs) promql.Sample {
```

Comment:
I wonder if it is worth adding this behind an option in the engine? Perhaps not and it will be available to everyone who knows about this :smile:

Reply:
Having an explicit option makes sense to me as well.